src/common/cockpitpipe.c

```diff
@@ -372,6 +374,24 @@ dispatch_input (gint fd,
   return TRUE;
 }
 
+static void
+forward_error (CockpitPipe *self)
+{
+  CockpitPipePrivate *priv = cockpit_pipe_get_instance_private (self);
+
+  guint8 *data = priv->err_buffer->data;
+  guint len = priv->err_buffer->len;
+  while (len > 0)
+    {
+      ssize_t n = TEMP_FAILURE_RETRY (write (2, data, len));
+      if (n <= 0)
+        break;
+      len -= n;
```

Comment:
`g_byte_array_remove_range()` probably gives you a nicer way to implement this function

Reply:
That would mean some copies, no?

Anyway, what about just using `fputs` and ignoring errors? :-) That's what glib does by default, no?

Reply:
I have pushed a variant that uses fwrite.

Reply:
It would only mean copies in case of partial writes, which we're not expecting.